communicator/step_connect_ssh.go

```diff
@@ -50,7 +50,14 @@ func (s *StepConnectSSH) Run(ctx context.Context, state multistep.StateBag) mult
 	}()
 
 	log.Printf("[INFO] Waiting for SSH, up to timeout: %s", s.Config.SSHTimeout)
-	timeout := time.After(s.Config.SSHTimeout)
+	var timeout <-chan time.Time
+	// Only setup a timeout when it is specified in the configs
+	if s.Config.SSHTimeout != 0 {
+		timeout = time.After(s.Config.SSHTimeout)
+	} else {
+		timeout = make(<-chan time.Time)
+	}
+
```

Comment:
Nit: we can assign the default case and drop the else
```suggestion
	timeout := make(<-chan time.Time)
	if s.Config.SSHTimeout > 0 {
		timeout = time.After(s.Config.SSHTimeout)
	} 
```